server/components/fill_teams_test.js

```diff
@@ -0,0 +1,195 @@
+/**
+* Created by Sammy on 1/13/15
+**/
+
+var db = require('../api/models');
+var _ = require('lodash');
+var BPromise = require("sequelize/node_modules/bluebird");
+
+var Fill = (function() {
+  var stockTeams = [
+    {name: "Albany Alphas", id: 1, SportId: 1},
+    {name: "Alaska Arctics", id: 2, SportId: 1},
+    {name: "Baltimore Spirits", id: 3, SportId: 1},
+    {name: "Free Agency Team", id: 4, SportId: 1, special: 'freeagency'},
+    {name: "Commissioner Team", id: 5, SportId: 1, special: 'commish'}
+  ]
+
+  var stockPlayers = [
+    {name: "Rodger Umaechi",
+      realWorldTeam: "ALB",
+      id: 1,
+      SportId: 1},
+    {name: "Apiatan Redmane",
+      realWorldTeam: "ALB",
+      id: 2,
+      SportId: 1},
+    {name: "Carl Esteban",
+      realWorldTeam: "ALK",
+      id: 3,
+      SportId: 1},
+    {name: "Ambrose Slade",
+      realWorldTeam: "ALK",
+      id: 4,
+      SportId: 1},
+    {name: "Silvia Windcreek",
+    realWorldTeam: "BAL",
+      id: 5,
+      SportId: 1},
+    {name: "Dirk von Stryker",
+      realWorldTeam: "BAL",
+      id: 6,
+      SportId: 1},
+    {name: "Scoonie Barrett",
+      realWorldTeam: "HWI",
+      id: 7,
+      SportId: 1},
+    {name: "Alphonse Norwich IV",
+      realWorldTeam: "HWI",
+      id: 8,
+      SportId: 1},
+    {name: "Neil LaRocca",
+      realWorldTeam: "PLY",
+      id: 9,
+      SportId: 1},
+    {name: "Nina Lima",
+      realWorldTeam: "PLY",
+      id: 10,
+      SportId: 1}
+  ]
+
+  var stockSport = [
+    {name: "NBA",
+      id: 1}
+  ]
+
+  var currentSport = ""
+  var selectedTeams = ""
+  var selectedPlayers = ""
+  var realWorldTeams = []
+
+  // var _fillTeams = function(user) {
+  //   new BPromise(function(resolve, reject) {
+  //     return userRoleCheck(user);
+  //   }).then(function() {
+  //     return setUpDatabase();
+  //   }).then(function() {
+  //     return findTeams();
+  //   }).then(function() {
+  //     return assignPlayers(selectedTeams, selectedPlayers)
+  //   });
+  // }
+
+  var _fillTeams = function(user) {
```

Comment:
Managed to chop up the previous version of the functions into this one, and it looks much better already at a glance.